src/components/Pages/Search/index.js

```diff
@@ -14,33 +16,150 @@ import SearchResult from 'components/Pages/Search/searchResult.js'
 
 const SearchPage = () => {
 
-  const { searchIndex } = useRouteData();
+  const intl = useIntl();
+  let { searchIndex } = useRouteData();
+  let searchedTerm = ""
 
-  const title = "Search" // TODDO: ⚠️useIntl
+  useEffect(() => {
+    const input = document.getElementById("coa-search_input")
+    input.focus()
+    const filteredSearch = searchWorker(searchIndexWithUrl, input.value)
+    setSearchResults(filteredSearch)
+    if (typeof window !== 'undefined') {
+      window.location.hash = input.value.toLocaleLowerCase()
+    }
+  }, [searchIndex]);
+
+  // Check the url for a search term to apply
+  if (typeof window !== 'undefined' && window.location.hash.length > 1) {
+    // TODO: ...when filters are added to the url, create an array instead and use the '&' standard.
+    // const queryArr = decodeURIComponent(window.location.hash.split("#")[1]).split("&")
+    // searchedTerm = queryArr[0]
+    searchedTerm = decodeURIComponent(window.location.hash.split("#")[1])
+  }
+
+  // hook makes our input dynamic (useful for "as-you-type" filtering)
+  let [ searchString, setSearchString ] = useState(searchedTerm)
+
+  // Don't show pages without Url (there seems to be some pages that are 'live', but without a url - catch those here...)
+  const searchIndexWithUrl = searchIndex.filter( page => page.janisUrls.length > 0)
+
+  let [ searchResults, setSearchResults ] = useState(searchIndexWithUrl)
+
+  const searchButtonPressed = function() {
+    loader.start({
+      contentId: 'coa-search_results',
+      loaderId: 'coa-search_loading_wheel',
+    })
+    setTimeout(function(){
+      if (typeof window !== 'undefined') {
+        window.location.hash = searchString.toLocaleLowerCase()
+      }
+      const filteredSearch = searchWorker(searchIndexWithUrl, searchString)
+      setSearchResults(filteredSearch)
+      loader.end()
+    },1000)
+  }
+
+  const searchKeyInput = function(event) {
+    setSearchString(event.target.value)
+
+    // For Quick Search (no-delay)... use 👇this instead of 👆.
+    // const filteredSearch = searchWorker(searchIndexWithUrl, event.target.value)
+    // setSearchResults(filteredSearch)
+  }
```

Comment:
This could later be used as an "auto-complete" for searching within the nav in the future.